worker/dbaccessor/tracker.go

```diff
@@ -240,11 +247,10 @@ func (w *trackedDBWorker) loop() error {
 // ensureDBAliveAndOpenNewIfRequired is a bit long-winded, but it is a way to
 // ensure that the underlying database is alive and well. If it is not, we
 // attempt to open a new one. If that fails, we return an error.
-func (w *trackedDBWorker) ensureDBAliveAndOpenIfRequired(db *sql.DB) (*sql.DB, error) {
+func (w *trackedDBWorker) ensureDBAliveAndOpenIfRequired(ctx context.Context, db *sql.DB) (_ *sql.DB, err error) {
 	// Allow killing the tomb to cancel the context,
 	// so shutdown/restart can not be blocked by this call.
 	ctx, cancel := context.WithTimeout(context.TODO(), time.Second*10)
```

Comment:
```suggestion
	ctx, cancel := context.WithTimeout(ctx, time.Second*10)
```

Reply:
We want to tie the parent context to this context, so that if the tomb is dying, then we want to stop processing this function.

Reply:
Yes sorry, didn't pay enough attention. Fixed.